src/app.js

```diff
@@ -1,12 +1,88 @@
+import axios from "axios";
+import distance from "euclidean-distance";
+
+// Get the API token
+async function getToken() {
+  let token;
+  try {
+    const response = await axios.post(
+      "https://blue-bottle-api-test.herokuapp.com/v1/tokens",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    token = response.data.token;
+  } catch (e) {
+    console.error(`Exception occurred while fetching the token: ${e}`);
+  }
+  return token;
+}
+
+// Get all shop data from the API
+async function getShopData() {
+  const token = await getToken();
+
+  try {
+    const response = await axios.get(
+      `https://blue-bottle-api-test.herokuapp.com/v1/coffee_shops?token=${token}`,
+      {
+        headers: {
+          Accept: "application/json",
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    if (e.response.status == 401 || 406 || 503 || 504) {
+      console.error(`Exception occurred while fetching shop list : ${e}`);
+      return [];
+    }
+  }
+}
+
 /**
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
-export function getNearestShops(position) {
-  // code
-  
-  return [];
+
+// Calculate distance of each shop from the given coordinate
+export default async function getNearestShops(position) {
+  const shopData = await getShopData();
+
+  if (shopData.length > 0) {
```

Comment:
This condition is not necessary.
Also, I think you'll enjoy this short article: https://jesseduffield.com/Array-Functions-And-Rule-Of-Least-Power/

Reply:
Yes, you are right. I just wanted to let the user know with an error message in case the API returns an error or an empty array.

Thanks for the link! I will be reading it and I will be glad to know more about it.